selenium/src/test/java/com/deque/html/axecore/selenium/WebDriverExtensionsTest.java

```diff
@@ -0,0 +1,152 @@
+      exception = e;
+    }
+
+    // store exception and the current url in the webDriver to be checked later
+    ArrayList<Object> exceptionAndUrl = new ArrayList<>();
+
+    exceptionAndUrl.add(exception);
+    exceptionAndUrl.add(webDriver.getCurrentUrl());
+
+    webDriver.quit();
+
+    return exceptionAndUrl;
+  }
+
+  @Test
+  public void shouldNotThrowGivenChromedriver() {
+    ArrayList<Object> exceptionAndUrl = tryOpenAboutBlank(Browser.CHROME);
+
+    Exception exception = (Exception) exceptionAndUrl.get(0);
+    String url = (String) exceptionAndUrl.get(1);
+
+    Assert.assertNull(exception);
+    Assert.assertEquals(url, "about:blank");
+  }
+
+  @Test
+  public void shouldNotThrowGivenGeckodriver() {
+    ArrayList<Object> exceptionAndUrl = tryOpenAboutBlank(Browser.FIREFOX);
+
+    Exception exception = (Exception) exceptionAndUrl.get(0);
+    String url = (String) exceptionAndUrl.get(1);
+
+    Assert.assertNull(exception);
+    Assert.assertEquals(url, "about:blank");
+  }
+
+  @Test
+  public void shouldNotThrowGivenSafariDriver() {
+    // if OS is windows or linux, skip this test as Safari is not available
+    String os = System.getProperty("os.name").toLowerCase();
+    Assume.assumeFalse(os.contains("windows"));
+    Assume.assumeFalse(os.contains("linux"));
+
+    ArrayList<Object> exceptionAndUrl = tryOpenAboutBlank(Browser.SAFARI);
+
+    Exception exception = (Exception) exceptionAndUrl.get(0);
+    String url = (String) exceptionAndUrl.get(1);
+
+    Assert.assertNull(exception);
+    Assert.assertEquals(url, "about:blank");
+  }
+
+  @Test
+  public void shouldThrowWhenSwitchToFails() {
+    // Create a mock driver to throw an exception when switchTo() is called
+    // This is to simulate `switchTo()` failing and throwing an exception
+    // We expect the exception to be caught and handled correctly.
+    class MockedDriver extends ChromeDriver {
+      public MockedDriver(ChromeOptions chromeOptions) {
+        super(chromeOptions);
+      }
+
+      @Override
+      public WebDriver.TargetLocator switchTo() {
+        throw new RuntimeException("BOOM!");
+      }
+    }
+
+    MockedDriver webDriver = new MockedDriver(new ChromeOptions().addArguments("--headless"));
+    webDriver.get("http://localhost:8001/index.html");
+
+    Exception exception =
+        Assert.assertThrows(
+            Exception.class,
+            () -> {
+              WebDriverExtensions.openAboutBlank(webDriver);
+            });
+
+    Assert.assertTrue(exception.getMessage().contains("switchToWindow failed."));
+  }
+
+  @Test
+  public void shouldThrowWhenUnableToDetermineWindowHandle() {
+    class MockedDriver extends ChromeDriver {
+      public MockedDriver(ChromeOptions chromeOptions) {
+        super(chromeOptions);
+      }
+
+      @Override
+      public Object executeScript(String script, Object... args) {
+        super.executeScript(script, args);
+        return super.executeScript(script, args);
```

Comment:
Is the purpose of this to call the open window twice so there are two new handles? A comment would be nice describing that if so as I had to think about this code for a bit.

Reply:
Yes, sorry! Will add a comment, this is to simulate the case where another window gets created other than `about:blank`. An edge case, but a good one to cover and get a identifiable error message for.